core/src/main/java/org/elasticsearch/index/query/IndicesQueryBuilder.java

```diff
@@ -31,48 +36,52 @@
 
     public static final String NAME = "indices";
 
-    private final QueryBuilder queryBuilder;
+    private final QueryBuilder innerQuery;
 
     private final String[] indices;
 
-    private String sNoMatchQuery;
-    private QueryBuilder noMatchQuery;
+    private QueryBuilder noMatchQuery = QueryBuilders.matchAllQuery();
```

Comment:
maybe have a static package private method that returns the default recreating a new match_all all the time. That could be used from the parser and below in the setter too?